### DISCUSSION TITLE
set default attr casing

### DISCUSSION THREAD

**Discussion by mesqueeb** [vuejs/language-tools#1714] [Q&A] [answered]
Volar gives this:
![image](https://user-images.githubusercontent.com/3253920/184338430-390a7ec9-409f-4d3a-ab13-1ddf99ca78f0.jpeg)

How do I set it to default to `camelCase` in the `.vscode/settings.json` ?

**Comment by kingyue737** [comment, marked as answer]
Are you looking for this?
![image](https://user-images.githubusercontent.com/40021217/184798650-028c993b-ca51-4fa3-985f-a972132b3cfa.png)

```json
{
  "volar.completion.preferredAttrNameCase": "auto-camel"
}
```

**Reply by mesqueeb** [reply]
@kingyue737 what's the difference between auto-camel and camel?

Also, do you know if there is any official documentation on these options?
I can't find much on the Github Repo.

**Reply by kingyue737** [reply]
There is hint shown in combo box. I think `camel` will force auto-completion of props to camel-case even if all the props already existing in the component are written in kebab-case, while `auto-camel` will still auto-complete kebab-case for you in this component. 

![image](https://user-images.githubusercontent.com/40021217/186350435-4298c1f1-bbce-4427-bdd6-7e4a46de528f.png)

From my test, you may need to restart volar server to make the setting effective.

**Reply by kingyue737** [reply]
You are right. Currently, there is no documentation on these options. I guess that the official documentation site will be released with v1.0.0 as it already exists in [docs](https://github.com/johnsoncodehk/volar/tree/master/docs) folder but far from finishing.
